fault-tolerance/src/main/java/io/helidon/faulttolerance/Retry.java

```diff
@@ -640,6 +641,387 @@ public Builder config(Config config) {
         }
     }
 
+    /**
+     * A retry policy that increases the delay time following the Fibonacci sequence.
+     * Allowed elements that are also annotated with {@code @Retry}.
+     * Expected sequence: initial delay, 2 * initial delay + jitter, 3 * initial delay + jitter,
+     * 5 * initial delay + jitter, etc. {@code maxDelayInMillis} is used to prevent endless waiting.
+     */
+    class FibonacciRetryPolicy implements RetryPolicy {
+
+        private final int calls;
+        private final Duration initialDelay;
+        private final Duration maxDelay;
+        private final Supplier<Long> randomJitter;
+
+        private long delayA;
+        private long delayB;
+
+        private FibonacciRetryPolicy(Builder builder) {
+            calls = builder.calls;
+            initialDelay = builder.initialDelay;
+            maxDelay = builder.maxDelay;
+            long jitter = builder.randomJitter;
+            int jitterMillis = (int) Math.min(jitter, Integer.MAX_VALUE);
+            if (jitterMillis == 0L) {
+                randomJitter = () -> 0L;
+            } else {
+                Random random = new Random();
+                // need a number [-jitterMillis,+jitterMillis]
+                randomJitter = () -> Long.valueOf(random.nextInt(jitterMillis * 2) - jitterMillis);
+            }
+        }
+
+        /**
+         * A new fluent API builder to configure instances of {@link io.helidon.faulttolerance.Retry}.
+         *
+         * @return a new builder
+         */
+        public static Builder builder() {
+            return new Builder();
+        }
+
+
+        @Override
+        public Optional<Long> nextDelayMillis(long firstCallMillis, long lastDelay, int call) {
```

Comment:
It may be nice to add some tests that create a strategy and repeatedly call `nextDelayMillis` to verify _the sequence_.

Reply:
added tests.